src/components/LmsConfigurations/CornerstoneIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,256 @@
+    case 'BASE_SUBMIT': {
+      return {
+        ...state,
+        submitState: SUBMIT_STATES.DEFAULT,
+      };
+    }
+    default: {
+      return { state };
+    }
+  }
+}
+
+function CornerstoneIntegrationConfigForm({ enterpriseId, config }) {
+  const [state, dispatch] = React.useReducer(
+    configFormReducer,
+    {
+      submitState: SUBMIT_STATES.default,
+      error: null,
+      config: null,
+      invalidFields: {},
+    },
+  );
+  const [active, setActive] = useState(config?.active);
+
+  /**
+   * Creates a new third party provider configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  const createCornerstoneConfig = async formData => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('enterprise_customer', enterpriseId);
+    try {
+      const response = await LmsApiService.postNewCornerstoneConfig(transformedData);
+      return dispatch({
+        type: 'SET_CONFIG',
+        value: response.data,
+      });
+    } catch (error) {
+      return handleErrors(error);
+    }
+  };
+
+  const updateCornerstoneConfig = async (formData, configId) => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('enterprise_customer', enterpriseId);
+    try {
+      const response = await LmsApiService.updateCornerstoneConfig(transformedData, configId);
+      return dispatch({
+        type: 'SET_CONFIG',
+        value: response.data,
+      });
+    } catch (error) {
+      return handleErrors(error);
+    }
+  };
+
+  /**
+  * attempt to submit the form data and show any error states or invalid fields.
+  * @param {FormData} formData
+  */
+  const handleSubmit = async (formData, existingConfig) => {
+    dispatch({
+      type: 'PENDING',
+    });
+    // validate the form
+    const invalidFields = validateLmsConfigForm(formData, REQUIRED_CORNERSTONE_CONFIG_FIELDS);
+    if (!isEmpty(invalidFields)) {
+      dispatch({
+        type: 'INVALID',
+        value: invalidFields,
+      });
+      return;
+    }
+
+    if (existingConfig) {
+      const err = await updateCornerstoneConfig(formData, existingConfig.id);
+      if (err) {
+        dispatch({
+          type: 'ERROR',
+          value: err,
+        });
+      } else {
+        dispatch({
+          type: 'COMPLETE',
+        });
+      }
+    } else {
+      // ...or create a new configuration
+      const err = await createCornerstoneConfig(formData);
+
+      if (err) {
+        dispatch({
+          type: 'ERROR',
+          value: err,
+        });
+      } else {
+        dispatch({
+          type: 'COMPLETE',
+        });
+      }
+    }
+  };
+
+  let errorAlert;
+  if (state.error) {
+    errorAlert = (
+      <div className="form-group is-invalid align-items-left">
+        <StatusAlert
+          alertType="danger"
+          iconClassName="fa fa-times-circle"
+          title="Unable to submit config form:"
```

Comment:
Same thought here about the microcopy. Although if all of these match the other LMS forms, maybe a ticket to change all the LMS microcopy to meet standards is more sensible than changing here.

Reply:
They all just match the Django Admin helper text, but slightly modified for user legibility. I can improve it, but I'm not 100% sure what piece of the microcopy it doesn't match. I guess punctuation? Not using second person pronouns?